Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -1470,3 +1469,20 @@ extension PackageBuilder {
             }
     }
 }
+
+private extension Target.Dependency {
+    
+    var duplicateDetectionHash: Int {
```

Comment:
thanks @sjavora, how about returning a string instead of a hash? e.g. (may not compile):

```
var nameAndType: String {
switch self {
             case .target:
                return "target-\(self.name)"
             case .product:
                return "product-\(self.name)"
         }
}
```

Reply:
Thanks @tomerd, compiles just fine - updated 👍.